packages/graphql-typescript-definitions/src/index.ts

```diff
@@ -222,16 +293,17 @@ export class Builder extends EventEmitter {
       return;
     }
 
-    this.emit('end');
+    this.emit('docs:end');
   }
 
-  private async updateSchema() {
+  private async updateSchema(schemaPath: string) {
     try {
-      const schemaJSON = await readJSON(this.schemaPath);
-      this.schema = buildClientSchema(schemaJSON.data);
+      const schemaJSON = await readJSON(schemaPath);
+      const schema = buildClientSchema(schemaJSON.data);
+      await this.generateSchemaTypes(schemaPath, schema);
```

Comment:
i noticed we could generate types in the `updateSchema` call because we were doing that after the call anyways.

Reply:
I think that it is better to leave `generateSchemaTypes` out of this function and add it back to where they are both invoked, just to keep levels of abstraction more consistent across the individual task functions like this, and to make the main flows a bit more grokkable.

Reply:
one of the motivations was that it made the `Promise.all(...)` a lot cleaner. but i guess i could just add a new `updateSchemaAndGenerateTypes` wrapper function that calls both `updateSchema` and `generateSchemaTypes`.